crates/oxc_linter/src/rules/eslint/no_array_constructor.rs

```diff
@@ -104,6 +112,7 @@ fn test() {
         ("Array?.<Foo>();", None),
         ("Array?.(0, 1, 2);", None),
         ("Array?.(x, y);", None),
+        ("var Array; new Array;", None),
```

Comment:
if you are interested in contributing another PR, porting more test cases for this rule would be very helpful 🙏

Reply:
Of course! I'd be happy to help.